### ISSUE TITLE
poetry search fails with ValueError: Unable to parse ".000001".

### ISSUE THREAD

**Issue by DrPyser** [sdispater/poetry#637]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: archlinux
- **Poetry version**: 0.12.8

## Issue

When I try to run `poetry search <name of my package>`, I get an error:
```
[ValueError]   
Unable to parse ".000001".  
                              
Exception trace:
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /home/charles/.poetry/lib/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /home/charles/.poetry/lib/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /home/charles/.poetry/lib/poetry/_vendor/py3.7/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /home/charles/.poetry/lib/poetry/console/commands/search.py in handle() at line 20
   results = PyPiRepository().search(self.argument("tokens"), flags)
 /home/charles/.poetry/lib/poetry/repositories/pypi_repository.py in search() at line 207
   result = Package(hit["name"], hit["version"], hit["version"])
 /home/charles/.poetry/lib/poetry/packages/package.py in __init__() at line 40
   self._version = Version.parse(version)
 /home/charles/.poetry/lib/poetry/semver/version.py in parse() at line 200
   raise ValueError('Unable to parse "{}".'.format(text))

search [-N|--only-name] [--] <tokens> (<tokens>)...
```
The package I'm searching is mine, and is on a private repository. The problem is probably related to my package, since `poetry search` otherwise works with other packages(existing or not).

**Comment by sdispater** [comment]
From the error you see, Poetry tries to parse a `.000001` version which is not valid.

**Comment by DrPyser** [comment]
Right, and where does this version come from? My package version(the one I'm trying to install) is "0.0.1".

**Comment by sdispater** [comment]
The `search` command does not search in private repositories, only on PyPI. So the package you are trying to search must exist on PyPI with a bad version.

**Comment by DrPyser** [comment]
Ok, well I assure you that it's not the case. `pip search my-package-name` confirms that. The package name is `kioto-basic-utils`, if you want to try for yourself.

**Comment by peter-schmidbauer** [comment]
I have run into a related error: Running the command `poetry search beautifulsoup` crashes with the error:

```
[ParseVersionError]
Unable to parse "rev452".
```

Seems like that command is quite brittle and crashes on packages using custom version strings. In this case, the package `ElementSoup` seems to be the culprit.

**Comment by peter-schmidbauer** [comment]
@sdispater while I agree that `.000001` is not a PEP 440 compliant version, it is still possible for packages on PyPi to have such invalid version strings, and it is unacceptable for poetry to just break when one pops up. I propose that packages with invalid versions should just be ignored and left out of the search results, which is also what PEP 440 recommends: 

`Installation tools SHOULD ignore any public versions which do not comply with this scheme but MUST also include the normalizations specified below.`

**Comment by peter-schmidbauer** [comment]
This issue is still not fixed in the newest release candidate 1.0.0b1.

**Comment by sztomi** [comment]
This is a duplicate of https://github.com/sdispater/poetry/pull/1346 (or rather, the latter is a duplicate of this one). I filed a PR to fix this.
